stripe-core/src/main/java/com/stripe/android/core/networking/AnalyticsRequestFactory.kt

```diff
@@ -0,0 +1,129 @@
+    private val publishableKeyProvider: Provider<String>,
+    internal val defaultProductUsageTokens: Set<String> = emptySet()
+) {
+
+    /**
+     * Builds an Analytics request for the given [AnalyticsEvent],
+     * including common params + event-specific params defined in [AnalyticsEvent.params]
+     *
+     * @param additionalParams any extra parameters that should be sent with this event.
+     * Ensure this common parameters are not already included in [standardParams].
+     *
+     */
+    fun createRequest(
+        event: AnalyticsEvent,
+        additionalParams: Map<String, Any>
+    ): AnalyticsRequest {
+        return AnalyticsRequest(
+            params = createParams(event) + additionalParams,
+            headers = RequestHeadersFactory.Analytics.create()
+        )
+    }
+
+    /**
+     * Backwards compatible for events not inheriting [AnalyticsEvent].
+     *
+     * @see createRequest
+     */
+    @Deprecated("use {@link #createRequest(AnalyticsEvent, Map<String, Any>)}")
+    fun createRequest(
+        event: String,
+        additionalParams: Map<String, Any>
+    ): AnalyticsRequest {
+        return createRequest(
+            event = object : AnalyticsEvent {
+                override val eventName: String = event
+            },
+            additionalParams = additionalParams
+        )
+    }
+
+    private fun createParams(
+        event: AnalyticsEvent
+    ): Map<String, Any> {
+        return standardParams() + appDataParams() + event.params()
+    }
+
+    private fun AnalyticsEvent.params(): Map<String, String> {
+        return mapOf(FIELD_EVENT to this.toString())
+    }
+
+    private fun standardParams(): Map<String, Any> = mapOf(
+        FIELD_ANALYTICS_UA to ANALYTICS_UA,
+        FIELD_PUBLISHABLE_KEY to runCatching {
+            publishableKeyProvider.get()
+        }.getOrDefault(ApiRequest.Options.UNDEFINED_PUBLISHABLE_KEY),
+        FIELD_OS_NAME to Build.VERSION.CODENAME,
+        FIELD_OS_RELEASE to Build.VERSION.RELEASE,
+        FIELD_OS_VERSION to Build.VERSION.SDK_INT,
+        FIELD_DEVICE_TYPE to DEVICE_TYPE,
+        FIELD_BINDINGS_VERSION to StripeSdkVersion.VERSION_NAME,
+        FIELD_IS_DEVELOPMENT to BuildConfig.DEBUG
+    )
+
+    internal fun appDataParams(): Map<String, Any> {
+        return when {
+            packageManager != null && packageInfo != null -> {
+                mapOf(
+                    FIELD_APP_NAME to getAppName(packageInfo, packageManager),
+                    FIELD_APP_VERSION to packageInfo.versionCode
+                )
+            }
+            else -> emptyMap()
+        }
+    }
+
+    private fun getAppName(
+        packageInfo: PackageInfo?,
+        packageManager: PackageManager
+    ): CharSequence {
+        return packageInfo?.applicationInfo?.loadLabel(packageManager).takeUnless {
+            it.isNullOrBlank()
+        } ?: packageName
+    }
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    companion object {
```

Comment:
we have NetworkConstants.kt, is it possible to move some of them over?

Reply:
Since these are more analytics-specific, should we maybe have an `AnalyticsConstants` or similar?

Reply:
Added `AnalyticsFields` object. https://github.com/stripe/stripe-android/pull/4651/commits/f4f42dff45c25700ca5e15bd16caf7c3abe03776

Should be more specific than `NetworkConstants`. Also by encapsulating them in an `object` we prevent constants to be globally indexed, which is usually a good practice to keep the IDE's index as light as possible.